components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/web/cache-mediator/update-mediator.jsp

```
@@ -15,83 +15,89 @@
 
-    if(notNullChecker(cacheid)){
-        cacheMediator.setId(cacheid);
-    }
-    if(notNullChecker(cacheScope)){
-        cacheMediator.setScope(cacheScope.toLowerCase());
-    }
-    if(notNullChecker(cacheType)){
-        if(cacheType.equalsIgnoreCase("Collector")){
+    if (notNullChecker(cacheType)) {
+        if (cacheType.equalsIgnoreCase("Collector")) {
             cacheMediator.setCollector(true);
-        }
-        else{
+        } else {
             cacheMediator.setCollector(false);
         }
     }
-    if(notNullChecker(cacheTimeout)){
-        try{
-        cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
-        }
-        catch(NumberFormatException e){
+    if (notNullChecker(cacheTimeout)) {
+        try {
+            cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
+        } catch (NumberFormatException e) {
 
         }
     }
-    if(notNullChecker(maxMsgSize)){
-        try{
-        cacheMediator.setMaxMessageSize(Integer.parseInt(maxMsgSize));
-        }
-        catch(NumberFormatException e){
+    if (notNullChecker(maxMsgSize)) {
+        try {
+            cacheMediator.setMaxMessageSize(Integer.parseInt(maxMsgSize));
+        } catch (NumberFormatException e) {
 
         }
     }
-    if(notNullChecker(impType)){
+
+    if (notNullChecker(protocolType)) {
+        cacheMediator.setProtocolType(protocolType);
+    }
+
+    if (notNullChecker(methods)) {
+        cacheMediator.setHTTPMethodsToCache(methods);
+    }
+
+    if (notNullChecker(headersToExclude)) {
+        cacheMediator.setHeadersToExcludeInHash(headersToExclude);
+    }
+
+    if (notNullChecker(responseCodes)) {
+        cacheMediator.setResponseCodes(responseCodes);
+    }
+
+    if (notNullChecker(hashGen)) {
+        cacheMediator.setDigestGenerator(hashGen);
+    }
+
+    if (notNullChecker(maxSize)) {
         try {
             cacheMediator.setInMemoryCacheSize(Integer.parseInt(maxSize));
         } catch (NumberFormatException e) {
```

This is for the UI. The check to make sure it is integer is carried out in the mediator-util.js so the exception is ignored here.